fury/ui/elements.py

```diff
@@ -3172,30 +3250,30 @@ def cal_bounding_box(self, position):
             if y > max_y:
                 max_y = y
 
-        self._bounding_box_min = [min_x, min_y]
-        self._bounding_box_max = [max_x, max_y]
-        self._bounding_box_size = [max_x-min_x, max_y-min_y]
+        self._bounding_box_min = np.asarray([min_x, min_y], dtype="int")
+        self._bounding_box_max = np.asarray([max_x, max_y], dtype="int")
+        self._bounding_box_size = np.asarray([max_x-min_x, max_y-min_y], dtype="int")
 
         self._bounding_box_offset = position - self._bounding_box_min
 
-    def clamp_position(self, position):
-        """Clamps the given position according to the DrawPanel canvas.
+    def clamp_position(self, center=None):
+        """Clamp the given center according to the DrawPanel canvas.
 
         Parameters
         ----------
-        position : (float, float)
+        center : (float, float)
             (x, y) in pixels.
 
         Returns
         -------
-        new_position: ndarray(int)
-            New position for the shape.
+        new_center: ndarray(int)
+            New center for the shape.
         """
-        self.cal_bounding_box(position)
-        new_position = np.clip(self._bounding_box_min, [0, 0],
-                               self.drawpanel.size - self._bounding_box_size)
-        new_position = new_position + self._bounding_box_offset
-        return new_position.astype(int)
+        if center is None:
+            center = self.center
```

Comment:
same as above: `center = center or self.center` instead of the `if`

Reply:
If I use this, it throws the below error
`ValueError: The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()`.

Should I use something like `center = self.center if center is None else center` ??

Reply:
yes